.github/workflows/sync-registry.yaml

```diff
@@ -0,0 +1,17 @@
+on:
+  schedule:
+    - cron: "0 0 * * *" # Runs every midnight
+
+jobs:
+  publish:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Sync registry.k8s.io to ghcr.io/canonical
+        env:
+          DEST_CREDS: ${{ github.actor }}:${{ secrets.GITHUB_TOKEN }}
+        run: |
+          chmod +x ${GITHUB_WORKSPACE}/build-scripts/hack/sync-registry.sh
```

Comment:
we should not chmod +x during CI, we should just chmod +x the file and commit it.